gcloud-java-core/src/main/java/com/google/cloud/FieldSelector.java

```diff
@@ -63,34 +63,51 @@ private static String selector(List<? extends FieldSelector> required, FieldSele
+     * This method is not supposed to be used directly by users.
      */
     public static String selector(List<? extends FieldSelector> required, FieldSelector... others) {
       return selector(required, others, new String[]{});
     }
 
     /**
-     * Returns a composite selector given a number of fields and a container name. The string
-     * selector returned by this method can be used for field selection in API calls that return a
-     * list of resources. This method is not supposed to be used directly by users.
+     * Returns a composite selector given a number of resource fields and a container name. The
+     * string selector returned by this method can be used for field selection in API calls that
+     * return a  list of resources. This method is not supposed to be used directly by users.
      */
     public static String listSelector(String containerName, List<? extends FieldSelector> required,
         FieldSelector... others) {
       return "nextPageToken," + containerName + '(' + selector(required, others) + ')';
     }
 
     /**
-     * Returns a composite selector given a number of fields and a container name. This methods also
-     * takes an {@code extraResourceFields} parameter to specify some extra fields as strings. The
-     * string selector returned by this method can be used for field selection in API calls that
-     * return a list of resources. This method is not supposed to be used directly by users.
+     * Returns a composite selector given a number of resource fields and a container name. This
+     * method also takes an {@code extraResourceFields} parameter to specify some extra resource
+     * fields as strings. The string selector returned by this method can be used for field
+     * selection in API calls that return a list of resources. This method is not supposed to be
+     * used directly by users.
      */
     public static String listSelector(String containerName, List<? extends FieldSelector> required,
         FieldSelector[] others, String... extraResourceFields) {
       return "nextPageToken," + containerName + '('
           + selector(required, others, extraResourceFields) + ')';
     }
+
+    /**
+     * Returns a composite selector given a number of first level fields as strings, a number of
+     * resource fields and a container name. This method also takes an {@code extraResourceFields}
+     * parameter  to specify some extra resource fields as strings. The string selector returned by
+     * this method can be used for field selection in API calls that return a list of resources.
+     * This method is not supposed to be used directly by users.
+     */
+    public static String listSelector(String[] firstLevelFields, String containerName,
+        List<? extends FieldSelector> required, FieldSelector[] others,
+        String... extraResourceFields) {
+      Set<String> firstLevelStrings = Sets.newHashSetWithExpectedSize(firstLevelFields.length + 1);
```

Comment:
Fine as is.

Another option (proably not worth the efforts):
```java
    public static String listSelector(String[] firstLevelFields, String containerName,
        List<? extends FieldSelector> required, FieldSelector[] others,
        String... extraResourceFields) {
      Set<String> firstLevelStrings = Sets.newHashSetWithExpectedSize(firstLevelFields.length + 2);
      firstLevelStrings.addAll(Lists.asList("nextPageToken", firstLevelFields));
      firstLevelStrings.add(containerName);
      StringBuilder stBuilder = new StringBuilder();
      Joiner.on(',').appendTo(stBuilder, firstLevelStrings);
      stBuilder.append('(');
      stBuilder.append(selector(required, others, extraResourceFields));
      stBuilder.append(')');
      return stBuilder.toString();
}
```

Also, this method could call a special selector that accepts StringBuilder to appendTo.